python/lsst/afw/image/multiband.py

```diff
@@ -0,0 +1,1093 @@
+        return MultibandPixel(filters=filters, singles=singles, bbox=bbox)
+
+    def __getitem__(self, indices):
+        """Get a slice of the underlying array
+
+        Since a `MultibandPixel` is a scalar in the
+        spatial dimensions, it can only be indexed with
+        a filter name, number, or slice.
+        """
+        if isinstance(indices, tuple):
+            err = ("Too many indices: "
+                   "`MultibandPixel has no spatial dimensions and "
+                   "only accepts a filterIndex")
+            raise IndexError(err)
+        # Make the index into a valid numpy index or slice
+        filters, filterIndex = self.filterToIndex(indices)
+        if len(filters) == 1 and not isinstance(filterIndex, slice):
+            # The user only requested a pixel in a single band, so return it
+            return self.array[filterIndex[0]]
+
+        result = self.copy()
+        result._filters = filters
+        result._singles = self._singles[filterIndex]
+        # No need to update the bounding box, since pixels can only be sliced in the filter dimension
+        return result
+
+    def setBBox(self, bbox):
+        """Overload `MultibandBase.setBBox`
+
+        Since the "single band objects" are elements
+        of an array and do not have a bounding box,
+        overloading `setBBox` is needed to prevent the
+        code from attempting to set the bounding box of
+        each item in `singles`.
+
+        Parameters
+        ----------
+        bbox: Point2I
+            Pixel location in the parent image
+        """
+        if not isinstance(bbox, Point2I):
+            err = "The bounding box for a `MultibandPixel` should always be a `Point2I`, received {0}"
+            raise ValueError(err.format(bbox))
+        self._bbox = bbox
+
+
+class MultibandImage(MultibandBase):
+    """Multiband Image class
+
+    This class acts as a container for multiple `afw.Image` objects.
+    All images must be contained in the same bounding box,
+    and have the same data type.
+    """
+    def __init__(self, filters, singles=None, array=None, bbox=None,
+                 singleType=ImageF, filterKwargs=None, **kwargs):
+        """Initialize a `MultibandBase` object
+
+        Parameters
+        ----------
+        singles: list
+            A list of single band objects.
+            If `array` is not `None`, then `singles` is ignored
+        filters: list
+            List of filter names. If `singles` is an `OrderedDict`
+            then this argument is ignored, otherwise it is required.
+        array: 3D numpy array
+            Array (filters, y, x) of multiband data.
+            If this is used to initialize a `MultibandImage`,
+            either `bbox` or `singles` is also required.
+        singleType: class
+            Class of the single band objects.
+            This is ignored unless `singles` and `array`
+            are both `None`, in which case it is required.
```

Comment:
We discussed this already in a previous comment, this is just a note for record keeping. `singleType` is needed because `MultibandMask` inherits from `MultibandImage`.